watcher/lib/resources/resources.go

```diff
@@ -0,0 +1,377 @@
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	return nil
+}
+
+// DeleteAlertTarget resets monitoring alerts recipient.
+func (c *Client) DeleteAlertTarget() error {
+	c.Info("Deleting alert target.")
+	conf, err := c.getAlertmanagerConfig()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	// We expect "default" receiver to be present.
+	defaultReceiver, err := getDefaultReceiver(conf)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	// Reset its email config.
+	defaultReceiver.EmailConfigs = nil
+	err = c.updateAlertmanagerConfig(conf)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	return nil
+}
+
+// CreateAlert creates a new or updates an existing monitoring alert.
+func (c *Client) CreateAlert(alert Alert) error {
+	c.Infof("Creating alert: %s.", alert)
+	prometheusRule := c.newPrometheusRule(alert)
+	_, err := c.Rules.Create(prometheusRule)
+	if err == nil {
+		return nil
+	}
+	err = rigging.ConvertError(err)
+	if !trace.IsAlreadyExists(err) {
+		return trace.Wrap(err)
+	}
+	_, err = c.Rules.Update(prometheusRule)
+	if err != nil {
+		return trace.Wrap(rigging.ConvertError(err))
+	}
+	return nil
+}
+
+// DeleteAlert deletes specified monitoring alert.
+func (c *Client) DeleteAlert(name string) error {
+	c.Infof("Deleting alert: %v.", name)
+	err := c.Rules.Delete(name, nil)
+	if err != nil {
+		return trace.Wrap(rigging.ConvertError(err))
+	}
+	return nil
+}
+
+// getAlertmanagerConfig returns Alertmanager configuration.
+func (c *Client) getAlertmanagerConfig() (*Config, error) {
+	secret, err := c.Secrets.Get(alertmanagerSecretName, metav1.GetOptions{})
+	if err != nil {
+		return nil, trace.Wrap(rigging.ConvertError(err))
+	}
+	confBytes, ok := secret.Data[alertmanagerConfigFilename]
+	if !ok {
+		return nil, trace.NotFound("no alert manager config found")
+	}
+	conf, err := Load(string(confBytes))
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return conf, nil
+}
+
+// updateAlertmanagerConfig updates Alertmanager configuration.
+func (c *Client) updateAlertmanagerConfig(conf *Config) error {
+	c.Debugf("Updating alertmanager configuration file: %#v.", conf)
+	secret, err := c.Secrets.Get(alertmanagerSecretName, metav1.GetOptions{})
+	if err != nil {
+		return trace.Wrap(rigging.ConvertError(err))
+	}
+	confString, err := conf.String()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	secret.StringData = map[string]string{
+		alertmanagerConfigFilename: confString,
+	}
+	_, err = c.Secrets.Update(secret)
+	if err != nil {
+		return trace.Wrap(rigging.ConvertError(err))
+	}
+	return nil
+}
+
+// getDefaultReceiver returns receiver with the name "default" from the config.
+func getDefaultReceiver(conf *Config) (*Receiver, error) {
+	for _, r := range conf.Receivers {
+		if r.Name == "default" {
+			return r, nil
+		}
+	}
+	return nil, trace.NotFound("no default receiver")
+}
+
+// updateSMTPConfig updates SMTP configuration in the provided config.
+func updateSMTPConfig(conf *Config, addr, user, pass string) error {
```

Comment:
I would introduce a `deleteSMTPConfig(conf)` so deleting it would not be cryptic:
```go
updateSMTPConfig(conf, "", "", "")
```